src/meltano/core/tracking/environment.py

```diff
@@ -0,0 +1,124 @@
+"""Environment context for the Snowplow tracker."""
+
+from __future__ import annotations
+
+import ctypes
+import ctypes.wintypes
+import os
+import platform
+import uuid
+from collections import defaultdict
+from datetime import datetime
+from pathlib import Path
+from platform import system
+from typing import Any
+
+import psutil
+from backports.cached_property import cached_property
+from snowplow_tracker import SelfDescribingJson
+from structlog.stdlib import get_logger
+
+import meltano
+
+logger = get_logger(__name__)
+
+# This file is only ever created in CI when building a release
+release_marker_path = Path(__file__).parent / ".release_marker"
+
+
+class EnvironmentContext(SelfDescribingJson):
+    def __init__(self):
+        logger.debug(
+            f"Initializing '{type(self).__module__}.{type(self).__qualname__}'"
+        )
+        ci_markers = ('GITHUB_ACTIONS', 'CI')
+        super().__init__(
+            "iglu:com.meltano/environment_context/jsonschema/1-0-0",
+            {
+                "context_uuid": uuid.uuid4(),
+                "meltano_version": meltano.__version__,
+                "is_dev_build": not release_marker_path.exists(),
+                "is_ci_environment": any(os.environ.get(x, '') == 'true' for x in ci_markers),
```

Comment:
Do we need to worry about case sensitivity or getting a value `'1'` or `1`?

Maybe this is too clever but not knowing the exact form of the variable's value, something like this seems safer:

```suggestion
                "is_ci_environment": any(str(os.environ.get(x, '')).lower()[:1] in ["1", "t"] for x in ci_markers),
```

Reply:
I originally had pretty much that exact code, but swapped it out for the less-clever version thinking I should stick to something simple that works exclusively with GitLab CI and GitHub Actions. Though in that case the name is no longer accurate. I ought to use `GITLAB_CI` instead of `CI`, and should replace `is_ci_environment` with a string enum called `ci` that can be one of:

- `null`
- `"github_actions"`
- `"gitlab_ci"`

That way we can get more info, and extend it with more CI options by simply adding new enum values to newer versions of the schema.